apps/ui/src/components/Modal/LabelConfig.vue

```diff
@@ -0,0 +1,105 @@
+<script setup lang="ts">
+import { clone } from '@/helpers/utils';
+import { getValidator } from '@/helpers/validation';
+import { SpaceMetadataLabel } from '@/types';
+
+const props = defineProps<{
+  open: boolean;
+  initialState?: SpaceMetadataLabel;
+}>();
+
+const emit = defineEmits<{
+  (e: 'add', config: SpaceMetadataLabel);
+  (e: 'close'): void;
+}>();
+
+const form = ref(
+  props.initialState ? clone(props.initialState) : clone(generateDefaultState())
+);
+
+const definition = {
+  type: 'object',
+  title: 'Space',
+  additionalProperties: false,
+  required: ['name', 'color'],
+  properties: {
+    name: {
+      type: 'string',
+      title: 'Label name',
+      minLength: 1,
+      maxLength: 32,
+      examples: ['council']
+    },
+    description: {
+      type: 'string',
+      title: 'Description',
+      maxLength: 100,
+      examples: ['Explain about label']
+    },
+    color: {
+      type: 'string',
+      format: 'color',
+      title: 'Color',
+      examples: ['#FF0000']
+    }
+  }
+};
+
+const formErrors = computed(() => {
+  const validator = getValidator(definition);
+  return validator.validate(form.value, { skipEmptyOptionalFields: true });
+});
+
+const formValid = computed(() => {
+  return Object.keys(formErrors.value).length === 0;
+});
+
+watch(
+  () => props.open,
+  () => {
+    form.value = clone(props.initialState || generateDefaultState());
+  }
+);
+
+function generateDefaultState(): SpaceMetadataLabel {
+  return {
+    id: crypto.randomUUID().substring(0, 8),
```

Comment:
Unless we are limited at database level I'd just keep full UUID.

Reply:
I have seen people hitting this limit https://github.com/snapshot-labs/snapshot-sequencer/blob/master/src/ingestor.ts#L64C5-L65C1 while saving their big whitelists to their settings, so I thought we could save some chars with this. but yea not too necessary